src/util.c

```diff
@@ -429,10 +425,6 @@ jv jq_util_input_next_input(jq_util_input_state *state) {
       }
       value = jv_parser_next(state->parser);
       if (jv_is_valid(state->slurped)) {
-        // When slurping an input that doesn't have a trailing newline,
-        // we might have more than one value on the same line, so let's check
-        // to see if we have more data to parse.
-        has_more = jv_parser_remaining(state->parser);
```

Comment:
`jv_parser_remaining` somehow ended up returning > 0 when there was nothing remaning but 0 if there was 1 or more trailing whitespace characters?

Reply:
The problem of using remaining byte count for detecting more value existing to slurp is that it does not work for `--stream` where it sometimes has array/object closing value. E.g., `printf '[1]' | jq --stream --slurp`, `jv_parser_next` emits `[[0],1]` and `has_more` is 0, even if there is another remaining value `[0]`. I fixed the issue by changing `is_last` value so that it works both for non-stream and stream inputs.

Reply:
Ah now i see 👍